src/scripts/core/models/column.ts

```diff
@@ -14,85 +15,90 @@ export class Column extends Container{
+            }
+            else {
                 nonMatchCombo.unshift(target);
-                for(let combo = columnInspector+1; combo <= this.tokens.length-1; combo++) {
-                    if(!this.tokens[combo].matched) { 
+                for (let combo = columnInspector + 1; combo <= this.tokens.length - 1; combo++) {
+                    if (!this.tokens[combo].matched) {
                         break;
                     }
-                    for(let i = 0; i < nonMatchCombo.length; i++){
-                        nonMatchCombo[i].moveTo(combo-i);
+                    for (let i = 0; i < nonMatchCombo.length; i++) {
+                        nonMatchCombo[i].moveTo(combo - i);
                     }
                 }
             }
         }
+        matchCombo.forEach(token => {
+            sortTimeline.add(() => token.moveTo(matchCombo.indexOf(token)), 2);
+            sortTimeline.add(() => token.reveal(), 2);
+            sortTimeline.add(() => token.shuffleSkin(), 2);
+        })
+        sortTimeline.play();
     }
 
     private rebuildTokens(): void {
-        const matchedTokens: Token [] = [];
-        const unmatchedTokens: Token [] = [];
-        for(let i = 0; i < this.tokens.length; i++) {
+        const matchedTokens: Token[] = [];
+        const unmatchedTokens: Token[] = [];
+        for (let i = 0; i < this.tokens.length; i++) {
             const target = this.tokens[i]
-            if(target.matched) {
-                target.hide();
-                target.interactive = false;
+            if (target.matched) {
                 matchedTokens.push(target);
             }
-
             else {
                 unmatchedTokens.push(target);
             }
         }
         const newTokens = [...matchedTokens, ...unmatchedTokens]
-        for(var i = 0; i < this.columnSize; i++) {
-            newTokens[i]._verticalIndex = i;
+        for (var i = 0; i < this.columnSize; i++) {
+            newTokens[i].matched = false;
+            newTokens[i].verticalIndex = i;
         }
         this.tokens = newTokens;
     }
 
-    public hideMatches(): void {
-        this.tokens.forEach(token => {
-            if(token.matched) {
-                token.hide()
-            }
-        });
-    }
-
     public deactivateAllTokens(): void {
         this.tokens.forEach(token => { token.interactive = false });
     }
 
-    public activateUnMatchedTokens(): void {
-        this.tokens.forEach(token => {
-            if(!token.matched) {
-                token.interactive = true;
-            }
-        });
+    public activateAllTokens(): void {
+        this.tokens.forEach(token => { token.interactive = true });
     }
 
     public getToken(Y: number) {
         return this.tokens[Y];
     }
 
     public replaceAllTokens(newTokens: Token[]): void {
-        if(newTokens.length != this.tokens.length) {
+        if (newTokens.length != this.tokens.length) {
```

Comment:
Another small thing, in Typescript the `!=` and `==` operators are `!==` and `===` respectively

Reply:
Like previously mentioned, these are 'truthy' and 'falsy' definitely want to do the changes curtis mentioned here